### PR TITLE
Fix and re-enable vectorized String.indexOf on 64-bit Power

### PR DESCRIPTION
**Pull request by aviansie-ben**
When it was first added, the vectorized versions of the `intrinsicIndexOf` helpers for Power had to be disabled due to mysterious difficult-to-reproduce segfaults it appeared to be causing. These have since been tracked down to the code falsely assuming that the upper word of a 64-bit register containing a 32-bit value were zeroes, which is not guaranteed. This PR corrects that issue, fixes a tangentially related sign-extension issue with the same helpers, and re-enables the vectorized versions of these helpers on 64-bit Power.

### CODE REVIEW

File: runtime/compiler/p/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -11808,15 +11808,21 @@ static TR::Register *inlineIntrinsicIndexOf(TR::Node *node, bool isLatin1, TR::C
    // to each String.indexOf call is ' ', which is the most common whitespace character.
       {
       TR::Register *value = srm->findOrCreateScratchRegister();
+      TR::Register *zxTargetScalar = srm->findOrCreateScratchRegister();
+
+      // Since we're going to do a load followed immediately by a comparison, we need to ensure that
```
Comment (gita-omr) [inline runtime/compiler/p/codegen/J9TreeEvaluator.cpp]: So you think we only need to do that for the first character and not for the rest (in the vector loop) ?

Comment (aviansie-ben) [reply runtime/compiler/p/codegen/J9TreeEvaluator.cpp]: That's correct. Only the first comparison is done using a 32-bit scalar comparison operator. It doesn't matter what the upper bits of this register are when performing the vector comparison, as they will have been overwritten when the value is splatted into the vector register.

**Comment by gita-omr** [comment]
Jenkins test sanity aix,plinux jdk8,jdk11

**Comment by gita-omr** [comment]
@pshipton this is a very good performance improvement to String.indexOf on Power. We would like to get it into 0.14.

**Comment by pshipton** [comment]
@gita-omr no objections, as long as it is promptly backed out or fixed if it causes any further problems

**Comment by pshipton** [comment]
@gita-omr perhaps wait until after the split for 0.14 is completed on Monday, then deliver to the master branch. If all looks good after a few days, then deliver it to the 0.14 branch as well.

**Comment by gita-omr** [comment]
Jenkins test extended aix,plinux jdk8,jdk11

**Comment by gita-omr** [comment]
@aviansie-ben @zl-wang please follow the process @pshipton described above.

**Pull request merged**